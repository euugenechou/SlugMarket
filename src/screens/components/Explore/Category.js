// React imports 
import React, { Component } from "react";
import { View, Text, StyleSheet, Image } from "react-native";

class Category extends Component {
  render() {
    return (
      <View style = { styles.container }>
        <View style = {{ flex: 2 }}>
          <Image
            source = { this.props.imageUri }
            style = { styles.image }
          />
        </View>
        <View style = { styles.text }>
          <Text style = {{ fontWeight: '600' }}> { this.props.name } </Text>
        </View>
      </View>
    );
  }
}

export default Category;

const styles = StyleSheet.create({
  text: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  container: {
    height: 120,
    width: 130,
    marginLeft: 20,
    borderRadius: 3,
    borderWidth: 1,
    borderColor: "#dddddd",
    shadowOffset: { width: 0, height: 0 },
    shadowColor: "black",
    shadowOpacity: 0.2,
    elevation: 1,
    backgroundColor: 'white'
  },
  image: {
    flex: 1,
    width: null,
    height: null,
    resizeMode: "cover",
    borderWidth: 0.1,
    borderRadius: 3,
    shadowOffset: { width: 0, height: 0 },
    shadowColor: "black",
    shadowOpacity: 0.2
  },
});
