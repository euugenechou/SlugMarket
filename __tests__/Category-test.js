// __tests__/Category-test.js
import 'react-native';
import React from 'react';
import Category from '../src/screens/components/Explore/Category';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Category />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});