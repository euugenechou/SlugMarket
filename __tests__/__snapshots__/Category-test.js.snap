// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#dddddd",
      "borderRadius": 3,
      "borderWidth": 1,
      "elevation": 1,
      "height": 120,
      "marginLeft": 20,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "width": 130,
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <Image
      style={
        Object {
          "borderRadius": 3,
          "borderWidth": 0.1,
          "flex": 1,
          "height": null,
          "resizeMode": "cover",
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "width": null,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "600",
        }
      }
    >
       
       
    </Text>
  </View>
</View>
`;
